host/4/bullseye/amd64/base/host.Dockerfile

```diff
@@ -10,7 +10,7 @@ RUN BUILD_NUMBER=$(echo ${HOST_VERSION} | cut -d'.' -f 3) && \
     git clone --branch v${HOST_VERSION} https://github.com/Azure/azure-functions-host /src/azure-functions-host && \
     cd /src/azure-functions-host && \
     HOST_COMMIT=$(git rev-list -1 HEAD) && \
-    dotnet publish -v q /p:BuildNumber=$BUILD_NUMBER /p:CommitHash=$HOST_COMMIT src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj -c Release --output /azure-functions-host --runtime linux-x64 && \
+    dotnet publish -v q /p:BuildNumber=$BUILD_NUMBER /p:CommitHash=$HOST_COMMIT src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj -c Release --output /azure-functions-host --runtime linux-x64 --no-self-contained && \
     mv /azure-functions-host/workers /workers && mkdir /azure-functions-host/workers && \
```

Comment:
@fabiocav Would like to get your thought on this.
Looks like removing self contained reduces the size of image from @CooperLink 's investigations.
For blessed images, we can control the exact version of dotnet installed on the image, so it should be okay.
For custom images, this might mean customers installing a different version of dotnet for their own code will interfere with functions runtime code.
What do you think? Any other  scenarios to consider?